src/sentry/uptime/endpoints/project_uptime_alert_checks_index.py

```diff
@@ -171,72 +171,54 @@ def _make_eap_request(
-        row_dict: dict[str, int | str | float] = {}
-        for col_idx, col_name in enumerate(column_names):
-            result = column_values[col_idx].results[row_idx]
-            row_dict.update(self._extract_field_value(result, col_name, uptime_subscription))
-
-        # XXX: Translate the status from `failed` to `failed_incident` when the
-        # check is part of an incident.
-        #
-        # TODO(epurkhiseer): In the future this should likely be part of the
-        # serializer that handles translating the snuba data into the expected
-        # data for the frontend.
-        in_incident = row_dict["incident_status"] == IncidentStatus.IN_INCIDENT.value
-        if row_dict["check_status"] == "failure" and in_incident:
-            row_dict["check_status"] = "failure_incident"
-
-        return convert_dict_key_case(row_dict, snake_to_camel_case)
-
-    def _extract_field_value(
-        self, result, col_name: str, uptime_subscription: ProjectUptimeSubscription
-    ) -> Mapping[str, int | str | float]:
-        if result.HasField("val_str"):
-            return self._handle_string_field(result.val_str, col_name, uptime_subscription)
-        elif result.HasField("val_int"):
-            return {col_name: int(result.val_int)}
-        elif result.HasField("val_double"):
-            return self._handle_double_field(result.val_double, col_name)
-        return {}
-
-    def _handle_string_field(
-        self, value: str, col_name: str, uptime_subscription: ProjectUptimeSubscription
-    ) -> Mapping[str, int | str | float]:
-        if col_name == "uptime_subscription_id":
-            # map this back to the project uptime subscription id
-            return {
-                "project_uptime_subscription_id": uptime_subscription.id,
-                col_name: uptime_subscription.id,
-            }
-        return {col_name: value}
-
-    def _handle_double_field(self, value: float, col_name: str) -> Mapping[str, int | str | float]:
-        if col_name in ("scheduled_check_time", "timestamp"):
-            return {col_name: datetime.fromtimestamp(value).strftime("%Y-%m-%dT%H:%M:%SZ")}
-        return {col_name: value}
+    ) -> EapCheckEntry:
+        row_dict: dict[str, AttributeValue] = {
+            col_name: column_values[col_idx].results[row_idx]
+            for col_idx, col_name in enumerate(column_names)
+        }
+
+        return EapCheckEntry(
+            uptime_check_id=row_dict["uptime_check_id"].val_int,
+            uptime_subscription_id=uptime_subscription.id,
+            timestamp=datetime.fromtimestamp(row_dict["timestamp"].val_double),
+            scheduled_check_time=datetime.fromtimestamp(
+                row_dict["scheduled_check_time"].val_double
+            ),
+            check_status=row_dict["check_status"].val_str,
+            check_status_reason=row_dict["check_status_reason"].val_str,
+            http_status_code=row_dict["http_status_code"].val_int,
```

Comment:
TODO: I think this may be empty sometimes, I need to update the test to account for this